src/unix/stream.c

```diff
@@ -839,11 +840,36 @@ static void uv__write(uv_stream_t* stream) {
 #endif
   } else {
     do {
+      /* On some platforms, notably OSX, the sum of iov_len must not
+       * overflow a 32-bit integer.
+       */
+      size_t old_iov_len = iov[iovcnt - 1].iov_len;
```

Comment:
I'm not the original author (and the original bug was found / fixed in our fork 2 years ago), so some of my responses are also partially just guesses.

It needs to save / restore the value for the last iovec that we transmit. At this point, it's assuming all of the iov will be sent (so the restore will be a no-op, but perhaps saves complicating the logic with an extra conditional to check that the restore is unnecessary). In the for-loop below, it updates this value as it goes while searching for the cutoff point.